scalameta/inputs/shared/src/main/scala/scala/meta/inputs/Input.scala

```diff
@@ -2,33 +2,43 @@ package scala.meta
 package inputs
 
 import java.nio.charset.Charset
+
 import org.scalameta.adt.{Liftables => AdtLiftables}
 import org.scalameta.data._
 import org.scalameta.invariants._
 import scala.meta.common._
 import scala.meta.internal.inputs._
+import scala.meta.internal.io.PathIO
 import scala.meta.io._
 
 trait Input extends Optional with Product with Serializable with InternalInput {
   def chars: Array[Char]
   def text: String = new String(chars)
+  def syntax: String
+  def structure: String
```

Comment:
How about we defer this change to the upcoming 2.x PR? There are some things to decide (whether `toString` should return syntax or structure, how to get rid of the `syntax/structure` boilerplate) that I wouldn't want to rush for 1.8.0.